CoreFeatures/Database-postgresql/PostgresSchema/src/main/java/info/smart_tools/smartactors/database_postgresql/postgres_schema/AlterClauses.java

```diff
@@ -0,0 +1,145 @@
+import info.smart_tools.smartactors.database.database_storage.utils.CollectionName;
+import info.smart_tools.smartactors.database_postgresql.postgres_schema.indexes.IndexCreators;
+import info.smart_tools.smartactors.database_postgresql.postgres_schema.search.FieldPath;
+import info.smart_tools.smartactors.database_postgresql.postgres_schema.search.PostgresFieldPath;
+import info.smart_tools.smartactors.iobject.ifield_name.IFieldName;
+import info.smart_tools.smartactors.iobject.iobject.IObject;
+import info.smart_tools.smartactors.iobject.iobject.exception.ReadValueException;
+import info.smart_tools.smartactors.ioc.iioccontainer.exception.ResolutionException;
+import info.smart_tools.smartactors.ioc.ikey.IKey;
+import info.smart_tools.smartactors.ioc.ioc.IOC;
+import info.smart_tools.smartactors.ioc.key_tools.Keys;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * A set of methods to write some SQL statements during the collection/table altering.
+ */
+final class AlterClauses {
+
+    /**
+     * Private constructor to prevent instantiation.
+     */
+    private AlterClauses() {
+    }
+
+    /**
+     * Writes expression for fulltext column creation and update,
+     * i.e. "ALTER TABLE collectionName ADD COLUMN ...; UPDATE collectionName SET fulltext_english := to_tsvector('english', ...)".
+     * @param body SQL query body to write
+     * @throws IOException if failed to write the body
+     */
+    static void writeAddFulltextColumn(final Writer body, final CollectionName collection, final IObject options)
+            throws QueryBuildException, IOException {
+        if (options == null) {
+            // if no fulltext option just skip
+            return;
+        }
+        String language;
+        List<FieldPath> fields;
+        try {
+            IKey fieldKey = Keys.getKeyByName("info.smart_tools.smartactors.iobject.ifield_name.IFieldName");
+
+            IFieldName fulltextDefinitionField = IOC.resolve(fieldKey, "fulltext");
+            Object fulltextFields = options.getValue(fulltextDefinitionField);
+            if (fulltextFields == null) {
+                // if no fulltext option just skip
+                return;
+            }
+
+            IFieldName languageField = IOC.resolve(fieldKey, "language");
+            language = (String) options.getValue(languageField);
+            if (language == null) {
+                language = PostgresSchema.DEFAULT_FTS_DICTIONARY;
+            }
+
+            fields = new ArrayList<>();
+            if (fulltextFields instanceof String) {
+                fields.add(PostgresFieldPath.fromString((String) fulltextFields));
+            } else if (fulltextFields instanceof List) {
+                for (Object fieldName : (List) fulltextFields) {
+                    fields.add(PostgresFieldPath.fromString((String) fieldName));
```

Comment:
!!! If the list element is not a String, this raises ClassCastException which is not caught here. Need to add an explicit type checks for list items.

Reply:
done